platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java

```diff
@@ -60,130 +60,249 @@
-            Path exportDirectory) {
-        Stopwatch stopwatch = null;
+    @Override
+    public Collection<MigrationWarning> doImport(String exportDirectory) throws MigrationException {
+        Validate.notNull(exportDirectory, "invalid null export directory");
+        final MigrationReport report = doImport(Paths.get(exportDirectory));
 
-        if (LOGGER.isDebugEnabled()) {
-            stopwatch = Stopwatch.createStarted();
-        }
+        report.verifyCompletion(); // will throw error if it failed
+        return report.warnings()
+                .collect(Collectors.toList());
+    }
 
-        MigrationMetadata migrationMetadata = migratable.export(exportDirectory);
+    @Override
+    public MigrationReport doImport(Path exportDirectory) {
+        return doImport(exportDirectory, Optional.empty());
+    }
 
-        if (LOGGER.isDebugEnabled() && stopwatch != null) {
-            LOGGER.debug("Export time: {}",
-                    stopwatch.stop()
-                            .toString());
-        }
+    @Override
+    public MigrationReport doImport(Path exportDirectory, Consumer<MigrationMessage> consumer) {
+        Validate.notNull(consumer, "invalid null consumer");
+        return doImport(exportDirectory, Optional.of(consumer));
+    }
+
+    @VisibleForTesting
+    ImportMigrationManagerImpl delegateToImportMigrationManager(MigrationReportImpl report,
+            Path exportFile) {
+        final ImportMigrationManagerImpl mgr = new ImportMigrationManagerImpl(report,
+                exportFile,
+                migratables.stream());
 
-        return migrationMetadata.getMigrationWarnings();
+        report.record(Messages.IMPORTING_DATA, exportFile);
+        mgr.doImport(productVersion);
+        return mgr;
     }
 
-    private Collection<MigrationWarning> exportMigratables(Path exportDirectory)
+    @VisibleForTesting
+    void delegateToExportMigrationManager(MigrationReportImpl report, Path exportFile)
             throws IOException {
-        List<MigrationWarning> warnings = new LinkedList<>();
+        try (final ExportMigrationManagerImpl mgr = new ExportMigrationManagerImpl(report,
+                exportFile,
+                migratables.stream())) {
+            report.record(Messages.EXPORTING_DATA, exportFile);
+            mgr.doExport(productVersion);
+        }
+    }
 
-        for (ConfigurationMigratable configMigratable : configurationMigratables) {
-            warnings.addAll(exportMigratable(configMigratable, exportDirectory));
+    private MigrationReportImpl doExport(Path exportDirectory,
+            Optional<Consumer<MigrationMessage>> consumer, boolean timestamp) {
+        Validate.notNull(exportDirectory, "invalid null export directory");
+        final MigrationReportImpl report = new MigrationReportImpl(MigrationOperation.EXPORT,
+                consumer);
 
+        try {
+            FileUtils.forceMkdir(exportDirectory.toFile());
+        } catch (IOException e) {
```

Comment:
We can't since we are trying to get the exception traces as well and they do not support using the parametrized logging when passing in an exception to trace both its message and its stack trace.